### ISSUE TITLE
[A11Y][Text input] Incorrect reading order

### ISSUE THREAD

**Issue by pya35** [Orange-OpenSource/ouds-flutter#449]
### Prerequisites

- [x] I have [searched the backlog](https://github.com/Orange-OpenSource/ouds-flutter/issues) for duplicate or closed feature requests

### Your test device

Iphone

### OS Version

iOS 26

### App version

0.6.0

### Describe the issue

Talkback reads the helper text before the label, then prefix, suffix and finally the content (or placeholder if present).

### Expected Behavior

Talkback should read in order "role, label, prefix, content, suffix, helper text".

If input is empty but prefix/suffix present, don't read them, consider the field as empty.

If placeholder present, don't read it as a content (user shall understand that the field is empty), but as a hint.

**Comment by pya35** [comment]
@nouha06 also, on Android, the leading icon is not hidden, Talkback reads it "unlabelled". It should be hidden as it's decorative. 

If no label (for a search field for example), the input shall keep an accessible label

**Comment by nouha06** [comment]
@pya35  i have some questions:

1. When the status of the text input is disabled, when should the screen reader announce "disabled"? Should it be at the end?
Proposed order:
"role, label, prefix, content, suffix, helper text, disabled".
waht about placeholder ?
2. When the read-only status is enabled, what should the screen reader announce?

**Comment by pya35** [comment]
@nouha06 

1. disabled : "role, label, prefix, content, suffix, disabled, helper text" - as mentioned in the issue, if a placeholder is present, don't read it as a content (user shall understand that the field is empty), but as a hint, so Talkback/Voice Over will read it at last.

2. read-only : read it as disabled state

**Comment by nouha06** [comment]
The issue has been fixed and the ticket has been closed, as we cannot release a new version. That’s why the TestFlight blocking label was applied.

**Comment by B3nz01d** [comment]
Closed during Sprint # 23

**Comment by nouha06** [comment]
New TestFlight BETA upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: Design Toolbox Flutter (485)
Version: 0.6.0
Build type: BETA
Build details (GitHub): 451
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)
cc @pya35 (accessibility team)

**Comment by nouha06** [comment]
📣 New Firebase App Distribution BETA upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@Tayebsed93) to get access.

Display name: DesignToolbox (BETA) (485)
Version: 0.6.0
Build type: BETA
Build details (GitHub): 451
cc @B3nz01d (product owner)
cc @Tayebsed93 @AhmedAmineZr @nouha06 (dev team)
cc @jerome-regnier (design team)
cc @pya35 (accessibility team)

**Comment by pya35** [comment]
@nouha06 retested on 0.6.0 (520) on iOS, order is now correct except the placeholder which is read even when the input is not empty. Read the placeholder only if input is empty

**Comment by nouha06** [comment]
📣 New Firebase App Distribution Alpha upload available 🚀 (Android)

Please, get the new build from App Tester app, or contact the project maintainers (@AhmedAmineZr ) to get access.

Display name: DesignToolbox (ALPHA) (607)
Version: 1.0.0
Build type: Alpha
Build details (GitHub): 548
cc @B3nz01d (product owner)
cc @AhmedAmineZr @nouha06 (dev team)
cc @pya35 (accessibility team)

**Comment by nouha06** [comment]
New TestFlight Alpha upload available 🚀 (iOS)

Please, get the new build from TestFlight app, or contact the project maintainers (@AhmedAmineZr ) to get access.

Display name: Design Toolbox Flutter (607)
Version: 1.0.0
Build type: ALPHA
Build details (GitHub): 548
cc @B3nz01d (product owner)
cc @AhmedAmineZr @nouha06 (dev team)
cc @pya35 (accessibility team)

**Comment by pya35** [comment]
OK, placeholder issue is logged on #471

**Comment by B3nz01d** [comment]
Finished during Sprint # 26
